src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -174,7 +174,7 @@ private IStorageAccount StorageAccount
             if (account != null)
             {
                 // On the first attempt, this will make a network call to verify the credentials work.
-                await _storageCredentialsValidator.ValidateCredentialsAsync(account, cancellationToken);
+                await _storageCredentialsValidator.ValidateCredentialsAsync(account, cancellationToken, account == StorageAccount);
```

Comment:
I think this validator is also used for Dashboard accounts? If so, it would be primary if either StorageAccount or DashboardAccount (both require queues access).